modules/yieldmoSyntheticInventoryModule.js

```diff
@@ -23,7 +23,24 @@ export function init(config) {
           window.top.googletag = window.top.googletag || {};
           window.top.googletag.cmd = window.top.googletag.cmd || [];
         }
-        getAd(`${AD_SERVER_ENDPOINT}?${serialize(collectData(config.placementId, consentDataObj))}`, config);
+        ajax.ajaxBuilder()(`${AD_SERVER_ENDPOINT}?${serialize(collectData(config.placementId, consentDataObj))}`, {
+          success: (responceText, responseObj) => {
+            window.top.__ymAds = processResponse(responseObj);
+            const googletag = window.top.googletag;
+            googletag.cmd.push(() => {
+              if (window.top.document.body) {
+                googletagCmd(config, googletag);
+              } else {
+                window.top.document.addEventListener('DOMContentLoaded', () => googletagCmd(config, googletag));
+              }
+            });
+          },
+          error: (message, err) => {
+            if (err && err.response) {
```

Comment:
@ym-abaranov  are you sure that we should not `throw` if `err` is missing? 
What if `message` is present?
I'd throw `message` or empty error in that case.
Finally, I don't think someone will call error handler if no errors happened, so you need to throw in any case.

Reply:
Good catch, thanks, fixed